zebedee-cms/src/main/java/com/github/onsdigital/zebedee/session/service/NewSessionsServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+            throw new IOException("client has failed to retrieve the session");
+        }
+
+        Session session = createZebedeeSession(clientSession);
+        return session;
+    }
+
+    /**
+     * Get a {@link Session} from the provided {@link HttpServletRequest}.
+     *
+     * @param request the {@link HttpServletRequest} to get the session for.
+     * @return a {@link Session} instance if once exists returns null otherwise.
+     * @throws IOException for any problem getting a sesison from the request.
+     */
+    @Override
+    public Session get(HttpServletRequest request) throws IOException {
+        String token = RequestUtils.getSessionId(request);
+        return get(token);
+    }
+
+    /**
+     * Get a {@link Session} by it's ID.
+     *
+     * @param id the ID of the session to get,
+     * @return the {@link Session} instance if it exists and is not expired.
+     * @throws IOException for any problems getting the session.
+     */
+    @Override
+    public Session get(String id) throws IOException {
+        com.session.service.Session clientSession = client.getSessionByID(id);
+        
+        return createZebedeeSession(clientSession);
+    }
+
+    /**
+     * Find a {@link Session} assocaited with the user email.
+     *
+     * @param email the user email address to find.
+     * @return a {@link Session} for the requested email if it exists and is not expired. Return null otherwise.
+     * @throws IOException for any problems getting the session.
+     */
+    @Override
+    public Session find(String email) throws IOException {
+        com.session.service.Session clientSession = client.getSessionByEmail(email);
+
+        Session session = createZebedeeSession(clientSession);
+        return session;
+    }
+
+    /**
+     * Check if an active {@link Session} exists with the provided ID.
+     *
+     * @param id the session ID to find.
+     * @return true if an active session exists with this ID, false otherwise.
+     * @throws IOException problem checking the session exists.
+     */
+    @Override
+    public boolean exists(String id) throws IOException {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Check if the provided {@link Session} is expired.
+     *
+     * @param session the {@link Session} to check.
+     * @return true if expired, false otherwise.
+     */
+    @Override
+    public boolean expired(Session session) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Get the expiry date of the provided {@link Session}
+     *
+     * @param session the  {@link Session} to use.
+     * @return the sessions expiration date time as a {@link Date} instance.
+     */
+    @Override
+    public Date getExpiryDate(Session session) {
+        throw new UnsupportedOperationException();
+    }
+
+    private Session createZebedeeSession(com.session.service.Session clientSession) {
+        Session session = new Session();
```

Comment:
**Super nit picky point** but there is no point in creating this object until the validation checks below have passed.  Move it to after the 2 validation checks.

If it fails its wasted effort and memory usage. Granted this is really minuscule saving but stuff like this adds up easily and contributes to a unnecessarily larger memory footprint.